torch/_dynamo/guards.py

```diff
@@ -767,13 +767,22 @@ def id_ref(self, obj):
         return id(obj)
 
 
+stashed_first_fail_reason = None
+
+
 def guard_fail_hook(
-    guard_fn: GuardFn, code: types.CodeType, f_locals: Dict[str, object], last: bool
+    guard_fn: GuardFn,
+    code: types.CodeType,
+    f_locals: Dict[str, object],
+    first: bool,
+    last: bool,
 ) -> None:
     """
     called whenever a guard fails.
     """
-    if not guard_fn.guard_fail_fn and not last:
+    global stashed_first_fail_reason
```

Comment:
Keep this on TracingContext, let's stop adding globals :)

Reply:
sgtm. where do i find tracing context?

Reply:
torch._guards.TracingContext